src/classes/task.js

```diff
@@ -0,0 +1,38 @@
+module.exports = class Task {
+  constructor() {
+    this.tasks = JSON.parse(localStorage.getItem('tasks')) || [
+    ];
+  }
+
+  bindTaskListChange(callback) {
+    this.onTaskListChanged = callback;
+  }
+
+  commit(tasks) {
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+    this.onTaskListChanged(tasks);
+    window.location.reload();
+  }
+
+  sort(curr, drop) {
+    const current = this.tasks[curr];
+    const replaced = this.tasks[drop];
+
+    current.index = drop;
+    replaced.index = curr;
+    this.tasks.splice(drop, 1, current);
+    this.tasks.splice(curr, 1, replaced);
+    this.commit(this.tasks);
+  }
+
+  addTask(taskText) {
+    const task = {
+      index: this.tasks.length > 0 ? this.tasks[this.tasks.length - 1].index + 1 : 0,
+      description: taskText,
+      completed: false,
```

Comment:
* Great work on the project so far 👏 but there is still one feature that needs implementing 
* upon user actions (ie drag and drop) the object value's index and completed status changes, currently only the completed status changes.
* when a list is dragged it should take along its index and completed status to its new position
* THIS is an image of a task in its original state : 
![image](https://user-images.githubusercontent.com/52497246/126521301-9cb897f4-3473-417b-9ddb-2e649f740e8c.png)
* This is an image of a task that has been dragged to a different location
* here you can see it doesn't retain its original index[2] but only retains its completed status[false] kindly fix this as it is a required feature [Refer here to project requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#:~:text=Update%20items%20object%27s%20values%20for%20index%20and%20completed%20keys%20upon%20user%20actions)
![image](https://user-images.githubusercontent.com/52497246/126521773-c697769d-fd4b-4777-b333-61f7dc58f2d6.png)